### PR TITLE
Differents Test for metric command added.

### PR DESCRIPTION
**Pull request by grihey**
Different tests added to check metric command work fine. Tests works with both existing and new metrics.

### CODE REVIEW

**Comment by unrealwork** [comment]
You should to get changes from base repository, we migrate to new test engine.

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    String existingMetricName = "metric-3088";
+
+    /* #3088 */
+
+    /*
+     * Test sets all possible parameters in a command.
+     * Also creates metric which will use like existing in other tests. Its reason why uderscore in method name.
+     */
+    @Test
+    public void _testOverall() throws Exception {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: what mean overall? what's exactly this method do?

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        sb.append(" p:").append(metric.getDataType().toString().toLowerCase());
+        sb.append(" v:").append(metric.getVersioned().toString());
+
+        for(Map.Entry<String, String> entry : tags.entrySet()) {
+            sb.append(" t:").append(entry.getKey() + "=" + entry.getValue());
+        }
+
+        tcpSender.send(sb.toString(), 1000);
+
+        Thread.sleep(1000);
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: you already specify sleep at `tcpSender.send()`

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        Response response = queryMetric(existingMetricName);
+
+        if (response.getStatus() != OK.getStatusCode()) {
+            response.close();
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: please use `bufferedEntity()` instead. It's common solution for any case

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+            throw new IOException("Fail to execute metric command.");
+        }
+
+        String actual = response.readEntity(String.class);
+        String expected = "{\"name\":\"" + metric.getName() + "\",\"dataType\":\"" + metric.getDataType().toString() +
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: don't use magic, we got Utility methods for this.
look at `jacksonMapper` and `compareJsonString()` instead

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+        }
+        expected = expected.substring(0, expected.length()-1);
+        expected += "}}";
+
+        if(!compareJsonString(expected, actual, false)) {
+            throw new IOException("Test is not passed.");
+        }
+    }
+
+    /* #3088 */
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: please look at naming convention example for test's comments:  https://github.com/axibase/atsd-api-test/blob/master/src/test/java/com/axibase/tsd/api/method/sql/metadata/SqlMetaDataTest.java#L50

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+        StringBuffer sb = new StringBuffer("metric");
+        sb.append(" m:").append(existingMetricName);
+
+        for(Map.Entry<String, String> entry : tags.entrySet()) {
+            sb.append(" t:").append(entry.getKey() + "=" + entry.getValue());
+        }
+
+        tcpSender.send(sb.toString(), 1000);
+
+        Thread.sleep(1000);
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+        Response response = queryMetric(existingMetricName);
+
+        if (response.getStatus() != OK.getStatusCode()) {
+            response.close();
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: `bufferedEntity()`

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+            throw new IOException("Fail to execute metric command.");
+        }
+
+        String actual = response.readEntity(String.class);
+        String expected = "{\"name\":\"" + existingMetricName + "\",\"tags\":{";
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: `jacksonMapper` & `comapreJsonString`

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+        expected += "}}";
+
+        if(!compareJsonString(expected, actual, false)) {
+            throw new IOException("Test is not passed.");
+        }
+    }
+
+    /* #3088 */
+    @Test
+    /*
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+        Thread.sleep(1000);
+        Response response = queryMetric(existingMetricName);
+
+        if (response.getStatus() != OK.getStatusCode()) {
+            response.close();
+            throw new IOException("Fail to execute metric command.");
+        }
+
+        String actual = response.readEntity(String.class);
+        String expected = "{\"name\":\"" + existingMetricName + "\",\"dataType\":\"" + dataType.toString() +
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+        StringBuffer sb = new StringBuffer("metric");
+        sb.append(" m:").append(metric.getName());
+
+        for(Map.Entry<String, String> entry : metric.getTags().entrySet()) {
+            sb.append(" t:").append(entry.getKey() + "=" + entry.getValue());
+        }
+
+        tcpSender.send(sb.toString(), 1000);
+
+        Thread.sleep(1000);
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 12]
```diff
@@ -0,0 +1,279 @@
...
+        Response response = queryMetric(name);
+
+        if (response.getStatus() != OK.getStatusCode()) {
+            response.close();
+            throw new IOException("Fail to execute metric command.");
+        }
+
+        String actual = response.readEntity(String.class);
+        String expected = "{\"name\":\"" + metric.getName() + "\",\"tags\":{";
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 13]
```diff
@@ -0,0 +1,279 @@
...
+        metric.setDataType(dataType);
+
+        StringBuffer sb = new StringBuffer("metric");
+        sb.append(" m:").append(metric.getName());
+        sb.append(" v:").append(metric.getVersioned());
+        sb.append(" p:").append(dataType.toString().toLowerCase());
+
+        tcpSender.send(sb.toString(), 1000);
+
+        Thread.sleep(1000);
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 14]
```diff
@@ -0,0 +1,279 @@
...
+        Response response = queryMetric(name);
+
+        if (response.getStatus() != OK.getStatusCode()) {
+            response.close();
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 15]
```diff
@@ -0,0 +1,279 @@
...
+            throw new IOException("Fail to execute metric command.");
+        }
+
+        String actual = response.readEntity(String.class);
+        String expected = "{\"name\":\"" + metric.getName() + "\",\"versioned\":" +
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 16]
```diff
@@ -0,0 +1,279 @@
...
+                "\"}}";
+
+        if(!compareJsonString(expected, actual, false)) {
+            throw new IOException("Test is not passed.");
+        }
+    }
+
+    /* #3088 */
+
+    /*
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 17]
```diff
@@ -0,0 +1,279 @@
...
+        StringBuffer sb =  new StringBuffer("metric");
+        sb.append(" m:").append(metric.getName());
+
+        for(Map.Entry<String, String> entry : metric.getTags().entrySet()) {
+            sb.append(" t:").append(entry.getKey() + "=" + entry.getValue());
+        }
+
+        tcpSender.send(sb.toString(), 1000);
+
+        Thread.sleep(1000);
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 18]
```diff
@@ -0,0 +1,279 @@
...
+        Response response = queryMetric(name);
+
+        if (response.getStatus() != NOT_FOUND.getStatusCode()) {
+            response.close();
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: -

Moreover, in this case you leave stream open. It's adduct to freeze of tests

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 19]
```diff
@@ -0,0 +1,279 @@
...
+     * Test checks metric with whitespaced tag creates or not.
+     */
+    @Test
+    public void testMalformedTag() throws Exception {
+        String name = "metric-malformed-1";
+        Map<String, String> tags = new HashMap<String, String>();
+        tags.put("m-ws tag","m-val-404");
+
+        Metric metric = new Metric(name);
+        metric.setTags(tags);
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: you able to use `metric.addTag()`

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java [hunk 20]
```diff
@@ -0,0 +1,279 @@
...
+
+    /* #3088 */
+    @Test
+    public void testCreateNewLongVersionedMetric() throws Exception {
+
+        String name = "metric-2";
+        Boolean versioned = true;
+        DataType dataType = DataType.LONG;
+
+        Metric metric = new Metric(name);
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCommandTest.java]: you can create metric directly: `new Metric("metric-2")`

**Comment by Sild** [comment]
end review

**Pull request closed without merging**